scylla-cql/src/frame/frame_errors.rs

```diff
@@ -67,6 +69,18 @@ pub enum ParseError {
     CqlTypeError(#[from] CqlTypeError),
 }
 
+/// An error type returned when deserialization of ERROR response fails.
+#[non_exhaustive]
+#[derive(Error, Debug)]
+pub enum CqlErrorParseError {
+    #[error("Malformed error code: {0}")]
+    ErrorCodeParseError(LowLevelDeserializationError),
+    #[error("Malformed error reason: {0}")]
+    ReasonParseError(LowLevelDeserializationError),
+    #[error(transparent)]
+    GenericDeserializationError(LowLevelDeserializationError),
```

Comment:
Maybe more context?

Reply:
```suggestion
    #[error("Malformed error field {field} of error with code {code}: {err})]
    MalformedErrorField {
        code: i32,
        field: &'static str,
        err: LowLevelDeserializationError,
    },
```

Reply:
As some errors are very rare, having plenty of information to identify the exact error and its problematic field can be invaluable in debugging.

Reply:
Ok, I agree. I'll add some context.